### ISSUE TITLE
Make it possible to deliver custom reporters via nuget packages

### ISSUE THREAD

**Issue by nvborisenko** [xunit/xunit#1681]
Currently reporter should be placed near `xunit.console.exe`. It would be really nice if reporter can be placed near test assembly. For end users it will mean that user can easily install nuget package into his project and execute tests via `xunit.console.exe`. So using 1 instance of runner user will be able to set reporters per test assembly.

**Comment by bradwilson** [comment]
Tagging as "Discussion" because I'm not sure what the best mechanism is here. We don't have access to any package information when tests are run.

**Comment by nvborisenko** [comment]
That's true. Currently xunit.console.exe loads reporters from working directory. Is it possible to include the directory where test assembly is located and try to load custom reporter from there?

**Comment by nvborisenko** [comment]
It would be nice that `visualstudio.adapter` finds reporters in the folder where test assembly located.

**Comment by nvborisenko** [comment]
@bradwilson it seems working solution. User installs custom reporter via nuget, and explicitly says to xunit.exe (console or adapter) to load this reporter via assembly attribute. There is no performance degradation due xunit.exe will not scan all assemblies in test directory.

**Comment by bradwilson** [comment]
Closing this due to the redesign of reporters in v3. They can now be shipped as a NuGet package, and then registered by the test project in question. More information: https://xunit.net/docs/getting-started/v3/custom-runner-reporter
